prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -48,8 +48,8 @@ object Parser extends StrictLogging {
   }
 
   // TODO: Once fully switched to AntlrParser, get rid of the special precedence methods.
-  private def parseQueryWithPrecedence(query: String): Expression = {
-    mode match {
+  private def parseQueryWithPrecedence(query: String, modeParam: Mode = mode): Expression = {
+    modeParam match {
```

Comment:
When does `mode`, the member come into play now with this change? Almost never right since it is always called with non-default ?